src/main/java/com/coherentsolutions/pot/insuranceservice/model/InsurancePackage.java

```diff
@@ -82,4 +81,19 @@ public class InsurancePackage {
   @Column(name = "updated_at")
   private Instant updatedAt;
 
+  public static PackageStatus calculateStatus(InsurancePackage insurancePackage,
+      boolean allowReactivation) {
+    if (!allowReactivation && insurancePackage.getStatus() == PackageStatus.DEACTIVATED) {
+      return PackageStatus.DEACTIVATED;
+    }
+    LocalDate now = LocalDate.now();
+    if (now.isBefore(insurancePackage.getStartDate())) {
+      return PackageStatus.INITIALIZED;
+    } else if (!now.isAfter(insurancePackage.getEndDate())) {
+      return PackageStatus.ACTIVE;
+    } else {
+      return PackageStatus.EXPIRED;
+    }
+  }
```

Comment:
This shouldn't be static now because you can perform this operation for the package you call this method on:
```suggestion
  public void calculateStatus(boolean allowReactivation) {
    if (!allowReactivation && this.status == PackageStatus.DEACTIVATED) {
      return;
    }
    var now = LocalDate.now();
    if (now.isBefore(this.startDate)) {
      this.status = PackageStatus.INITIALIZED;
    } else if (!now.isAfter(this.endDate)) {
      this.status = PackageStatus.ACTIVE;
    } else {
      this.status = PackageStatus.EXPIRED;
    }
  }
```
OOP :)